Automattic/vip-workflow-plugin
# Migrate EM to use term_meta

## Description

This PR migrates the storage system used by Editorial Metadata to be `term_meta` for the metadata items like `type` and `postmeta key`. This will set the stage for the same system to be used in custom status as well.

For now, the position ordering has been removed in favour of name based sorting. This is to keep things simple when it comes to the front end code. This will be needed for the custom status migration anyways, so we can duplicate that code then or when we add drag/drop support. This way, at least there's some order to the whole system.

I've also disabled the multisite tests as it was randomly causing 401 problems for the rest api. I've marked why we have disabled this, and this at least allows EM tests to be written. It seems like something to do with the user but we can check this out later.

## Steps to Test

- Delete all your terms before checking out this PR
- Delete the option for the module
- Check out this PR
- Allow it to do its magic as the storage schema has changed
- Ensure CRUD works fine
